### PR TITLE
Perception pipeline for April tags

### PR DESCRIPTION
**Pull request by HALtheWise**

### CODE REVIEW

File: iarc_vision/scripts/apriltag_rebroadcast.py [hunk 1]
```diff
@@ -1,4 +1,81 @@
...
+
+
+class TagTransformer(object):
+    """
+    A TagTransformer is responsible for transforming detections by the apriltags_ros library
+    into RoombaSighting messages for processing by the rest of the stack.
+    """
+    def __init__(self, linear_covariance=0.1, angular_covariance=0.5, tf_frame='map', camera_fov=1.0):
+        """
+        :param linear_covariance: (meters) the linear covariance associated with a tag detection
```
Comment (NathanYee) [inline iarc_vision/scripts/apriltag_rebroadcast.py]: There is a small inconsistency with where you place your ":" and the type

File: iarc_vision/scripts/apriltag_rebroadcast.py [hunk 2]
```diff
@@ -1,4 +1,81 @@
...
+from iarc_main.msg import RoombaSighting, Roomba
+
+
+class TagTransformer(object):
+    """
+    A TagTransformer is responsible for transforming detections by the apriltags_ros library
+    into RoombaSighting messages for processing by the rest of the stack.
+    """
+    def __init__(self, linear_covariance=0.1, angular_covariance=0.5, tf_frame='map', camera_fov=1.0):
+        """
```
Comment (NathanYee) [inline iarc_vision/scripts/apriltag_rebroadcast.py]: There are some small inconsistencies with the ":" and (type) placement.

File: iarc_vision/scripts/apriltag_rebroadcast.py [hunk 3]
```diff
@@ -1,4 +1,81 @@
...
+        :param linear_covariance: (meters) the linear covariance associated with a tag detection
+        :param angular_covariance: (radians) the angular covariance of a tag
+        :param tf_frame (str): The initial TF frame name to use before any detections come in.
+        :param camera_fov (float): The FOV radius of the camera, in meters.
```
Comment (NathanYee) [inline iarc_vision/scripts/apriltag_rebroadcast.py]: A little bit confused what fov is used for

File: iarc_vision/scripts/apriltag_rebroadcast.py [hunk 4]
```diff
@@ -1,4 +1,81 @@
...
+
+from iarc_main.msg import RoombaSighting, Roomba
+
+
+class TagTransformer(object):
+    """
+    A TagTransformer is responsible for transforming detections by the apriltags_ros library
+    into RoombaSighting messages for processing by the rest of the stack.
+    """
+    def __init__(self, linear_covariance=0.1, angular_covariance=0.5, tf_frame='map', camera_fov=1.0):
```
Comment (NathanYee) [inline iarc_vision/scripts/apriltag_rebroadcast.py]: why is 1m the default argument for camera_fov?

File: iarc_vision/scripts/apriltag_rebroadcast.py (hunk 1 above)
Comment (NathanYee) [inline iarc_vision/scripts/apriltag_rebroadcast.py]: Describe the default values in the docstring. These covariance values are arbitrary things we set right? When would we want to change them?

File: iarc_vision/scripts/apriltag_rebroadcast.py [hunk 5]
```diff
@@ -1,4 +1,81 @@
...
+            roomba = Roomba()
+            roomba.visible_location = PoseWithCovarianceStamped(
+                header=tag.pose.header,
+                pose=PoseWithCovariance(
+                    pose=tag.pose.pose,
+                    covariance=self.covariance
+                )
+            )
+
+            roomba.type = [Roomba.GREEN, Roomba.RED][tag.id % 2]
```
Comment (NathanYee) [inline iarc_vision/scripts/apriltag_rebroadcast.py]: comment something like:
\# assign even tags to green rombas and odd tags to red roombas

File: iarc_vision/scripts/apriltag_rebroadcast.py [hunk 6]
```diff
@@ -1,4 +1,81 @@
...
+
+            roomba.last_seen = tag.pose.header.stamp
+
+            sighting.data.append(roomba)
+            sighting.magical_ids.append(tag.id)
+
+            self.tf_frame = tag.pose.header.frame_id
```
Comment (NathanYee) [inline iarc_vision/scripts/apriltag_rebroadcast.py:80]: I'm not sure this is technically correct, but I don't really care.

**Review by NathanYee — APPROVED** [review]

**Comment by yycho0108** [comment]
Given the expected angular fov of about 90 deg. and drone height 1m~3m, 
FOV will probably be roughly between 2m - 6m. (2 * (h * tan(theta/2)))

Since the code isn't considering camera parameters, nor the drone's height,
setting the default as 1m is not a terrible approximation for right now --
even if the data processed based on this assumption will be worthless.

**Pull request merged**